color assignment tweaks

briangregoryholmes opened the pull request:
Please go the the `Preview` tab and select the appropriate sub-template:

* [Applications](?expand=1&template=applications_template.md)
* [Others](?expand=1&template=fallback_template.md)

nishantmonu51 commented:
@briangregoryholmes : Is this still a draft ? wondering if we can get this over the finish line for current release ?

briangregoryholmes commented:
@nishantmonu51 Took care of the last outstanding issue. Ready for review.

djbarnwal commented on web-common/src/features/dashboards/dimension-table/DimensionFilterGutter.svelte:
```diff
@@ -45,14 +45,22 @@
-          <CheckCircle className={getColor(row.index)} size="18px" />
+          <CheckCircle
+            className="fill-{colorGetter.get(
+              $page.params.name,
```
`metricViewName` should be available from `StateManagers`. We can probably use that to get the dashboard name and stay consistent with rest of the codebase?

djbarnwal commented on web-common/src/features/dashboards/filters/colorGetter.ts:
```diff
@@ -0,0 +1,74 @@
+    Map<DimensionName, Map<DimensionValue, number>>
+  >();
+
+  // Temporarily commenting out the derived nature because it's causing an issue with testing
+  // return derived([page], ([$page]) => {
+  //   const dashboardName = $page.params.name;
+
```
Are these stale comments?

djbarnwal commented on web-common/src/features/dashboards/filters/colorGetter.ts:
```diff
@@ -0,0 +1,74 @@
+import { CHECKMARK_COLORS } from "../config";
+// import { page } from "$app/stores";
+// import { derived } from "svelte/store";
+
```
Nit: Clean up

djbarnwal commented on web-common/src/features/dashboards/leaderboard/LeaderboardListItem.svelte:
```diff
@@ -42,6 +44,10 @@
 
+  $: checkColor = selected
+    ? colorGetter.get($page.params.name, dimensionName, label)
+    : "gray-300";
```
Use `metricViewName` from `StateManagers`

djbarnwal replied:
The default value returned is `gray-300` by `colorGetter`. Ternary operator might be redundant.

djbarnwal commented on web-common/src/features/dashboards/time-dimension-details/TDDTable.svelte:
```diff
@@ -115,8 +115,8 @@
-    scrubPos;
-    highlightedCol;
+    // scrubPos;
+    // highlightedCol;
```
This removes the hover effect on the column in TDD view when a chart is hovered over.

djbarnwal commented on web-common/src/features/dashboards/time-dimension-details/TimeDimensionDisplay.svelte:
```diff
@@ -87,6 +90,11 @@
 
+  $: markerColors =
+    values.map((value) =>
+      colorGetter.get($page.params.name, dimensionName, value),
```
Nit: Remove usage of `$page`

djbarnwal commented on web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte:
```diff
@@ -207,6 +209,10 @@
     showHideMeasures.setAllToVisible();
   };
+
+  $: colors = dimensionData.map((d) => {
+    return colorGetter.get($page.params.name, d.dimension, d.value ?? "");
+  });
```
Since this is only used for `MeasureChart`, we can move this inside the component.

briangregoryholmes replied:
It's placed here because the colors are not unique to each MeasureChart, but rather to the MetricsTimeSeriesChart view.  They only need to be fetched once regardless of the number of charts. Also, because of some upstream issues with TimeSeriesChartContainer and GraphicContext, MeasureChart will re-render dozens of times when you scrub or add new filter pills.

djbarnwal commented on web-common/src/features/dashboards/time-series/ChartBody.svelte:
```diff
@@ -84,20 +85,23 @@
+    {#each dimensionData as d, i}
       {@const isHighlighted = d?.value === dimensionValue}
+
+      {@const delay = 100}
```
Any reason for adding it here?

djbarnwal reviewed (changes requested):
Have added some comments around usage of `$page` and other changes.

Apart from that seeing some regression. When max colored comparison is reached, toggling off a dimension value doesn't change the colors of inactive dimensions.

https://github.com/rilldata/rill/assets/4402679/b9466249-b3fd-4a56-90fe-8665a65fabcd

Apart from that, for Leaderboard and Dimension Detail table, the max limit is 7 and for TDD it's 11 (since it has a bigger chart surface to accommodate more lines)

briangregoryholmes commented:
Updated to use metricsViewName throughout.

Your point about the regression is a fair one, though I'll rope in @mindspank @jkhwu for their thoughts since I think there are multiple solutions. In my view, we should never be reassigning a comparison color without explicit user interaction, even inactive ones. If I deselect blue, I expect the chart to no longer be showing a blue line.

As such, I updated the PR to match the "capping" behavior in a way that still maintains that behavior. Clicking an eighth comparison assigns it, for instance, `slate-500`. That check would display as `gray-300` since it's over the 7 comparison cap (on the Leaderboard), but if you deselect any of the other active colors, then it will turn to `slate-500`.

djbarnwal reviewed (commented):
Still seeing this state for TDD. Perhaps that was missed?

![image](https://github.com/rilldata/rill/assets/4402679/b047f53a-99d4-4760-9a8d-c558e3bccfe3)

nishantmonu51 commented:
@briangregoryholmes : bumping this up. Its open from a long time. Lets try to get this over the finish line this week. +cc @ericpgreen2

nishantmonu51 commented:
Closing this as stale, please reopen when ready for review. 
+cc @briangregoryholmes @ericpgreen2
